lib/msf/core/db_manager/http/servlet_helper.rb

```diff
@@ -13,21 +13,22 @@ def set_error_on_response(error)
   end
 
   def set_empty_response
-    [200,  '']
+    set_json_data_response(response: '')
   end
 
   def set_json_response(data, includes = nil, code = 200)
-    headers = {'Content-Type' => 'application/json'}
+    headers = { 'Content-Type' => 'application/json' }
     [code, headers, to_json(data, includes)]
   end
 
   def set_json_data_response(response:, includes: nil, code: 200)
-    data_response = {"data": response}
+    data_response = { "data": response }
     set_json_response(data_response, includes = includes, code = code)
   end
 
-  def set_json_error_response(response:, includes: nil, code:)
-    error_response = {"error": response}
+  def set_json_error_response(error:, includes: nil, code:)
```

Comment:
You shifted the meaning here a little from my original implementation. It would be nice to have a method where we could return an error response formatted by the client, rather than it being fixed to a message only.

Reply:
Not sure I understand. The "response" is the entire JSON message we are sending back. This takes the error message and creates the whole JSON response object. Actually, now that I look at it we should probably change `set_json_data_response` signature to be `(data: ...`.

Am I totally off base from what you're trying to say?

Reply:
The client of this method can only introduce a single message into the error message that is produced. What if I have additional data I want to include in the complete error response JSON? There could multiple errors, added under an "errors" object each with its own message and error code, etc.
```
error_response = { "error": { "message": error.message } }
```